### DISCUSSION TITLE
[2024-10-28] Libs and Engine Team Demo Day

### DISCUSSION THREAD

**Discussion by GregoryTravis** [enso-org/enso#11418] [Weekly Demo Day]
- James: Small fixes.. #11338
- James: Switch to newer MacOS GitHub runner #11345
- James: Handle mixed Date and Date Time column within Excel #11349
- James: Excel before 1900 and AWS signed requests. #11373
- James: Fixes for Google Analytics and widgets
- Radosław: Enable audit logs for Snowflake backend #11306
- Radosław: Suggest running Extra Tests if relevant changes are made in a PR #11315
- Radosław: Save Database connection as data link, SQL Server data link support #11343
- Radosław: Saving data links to a DB_Table #11371
- Radosław: Infer method calls on known types and warn about nonexistent methods #11399
- Radosław: Fixing small bugs uncovered by type checker #11422
- Greg: HTTP response caching, with TTL and LRU logic #11342
- Greg: Fix `Float.parse` benchmark regression #11402
- Greg: Clean up SQLite file after the SQLite file tests have run. #11416
- Greg: Add Warnings whenever a floating-point value is used as a Dictionary key #11417
- Ad: Update running tests documentation #11257
- Ad: Refactor Dialect_Flag #11273
- Ad: Set stack size to 16M #11327
- Ad: Enable SQLServer Sort Feature and associated tests #11379
- Ad: Change 3 dots icon #11400
- Martha: Display error message when Scatterplot Visualization errors out #11284
- Martha: Add data quality indicators to table Viz #11307
- Martha: Extend Selection Menu to allow text selection #11352
- Jaroslav: Avoid JavaScript and Python in Enso native launcher #11279
- Jaroslav: Special handling of last signature in a block #11289
- Jaroslav: Yield `UnsupportedMessageException` when context isn't accessible #11334
- Jaroslav: Benchmark lazy infinite sieve of Eratosthenes #11351
- Jaroslav: Benchmark time it takes compile `Standard.Base` library #11355
- Jaroslav: Make sure unrecognized CLI command message is printed #11357
- Jaroslav: Recognize if-like Tree.MultiSegmentApp as IfThenElse IR #11365
- Jaroslav: Simplify IGV graph by trimming nodes for code paths that rarely happen #11381
- Jaroslav: More robust enso4igv when "find in projects" invoked #11384
- Jaroslav: Use `enso.dev.insight` property to turn Insight on #11385
- Jaroslav: Skeletal PanicExceptionTest and more logging when AssertionError happens #11393
- Jaroslav: Execute `test/Geo_Tests` in enso binary launcher #11394
- Jaroslav: Testing current behavior of if/then/else #11395
- Jaroslav: Execute Base_Internal_Tests with native enso executable #11403
- Hubert: Handle autoscoped constructor args with no UUID #11354
- Hubert: Fail gracefully when a job is already rejected #11361
- Hubert: Don't cancel aborted jobs immediately #11375
- Hubert: Increase backend log-level for e2e tests on Mac #11367
- Hubert: Ensure wrapper projects are listed in aggregates #11323
- Pavel: [PoC] Generate IR definitions with annotation processor #11267
- Pavel: Remove TailCall.TailPosition.NotTail metadata. #11303
- Pavel: Fix invocation of single benchmarks in `std-benchmarks` #11321
- Pavel: Inline doc comment is a compiler error #11333
- Pavel: Set live execution environment for REPL #11353
- Pavel: Convert `Array_Like_Helpers.map` to a builtin to reduce stack size #11363
- Pavel: Fix Meta.get_qualified_type_name when run as single file #11401
- Dmitry: Node added inside collapsed node is never executed #11341
- Dmitry: Unblock the hooks execution #11359
- Dmitry: Metadata should not depend on absolute text spans #11390
- Dmitry: Reload insight script on file change #11415

**Comment by radeusgd** [comment]
This is how the datalinks look like in the GUI:

- Saving a table directly stores it by-name:
```
t.save_as_data_link (Enso_File.home / "postgres-table.datalink")
```
![image](https://github.com/user-attachments/assets/dc2c03d5-aaff-4c55-8b9f-bbf08bff2811)

- Simple query without interpolations (this could also be typed in by the user creating the datalink inside of the GUI):
```
t.rename_columns ["Y"] . save_as_data_link (Enso_File.home / "postgres-simple-query.datalink")
```
![image](https://github.com/user-attachments/assets/ea9cb482-7a5c-4736-a3de-5404583903ae)

- More complex serialized queries - these are not meant to be typed in by the user, but we still need to display them somehow. 
```
t.set (t.at "X" * 1000 + 45) "Z" . save_as_data_link (Enso_File.home / "postgres-query.datalink")
```
![image](https://github.com/user-attachments/assets/4e30ac21-b8f3-4e2d-8a12-4d97bca7acf5)

**Reply by radeusgd** [reply]
I've also added a warning if the query that is being saved refers to temporary tables - as after a session is closed those are destroyed and the datalink will most likely become invalid.

![image](https://github.com/user-attachments/assets/72170007-3f77-43e2-a6e2-99aa373f7126)
and after pressing the play button:
![image](https://github.com/user-attachments/assets/2cbd8b46-617b-421f-ba2b-11f46afd0935)
